### ISSUE TITLE
SideBar In a container

### ISSUE THREAD

**Issue by Olmer1st** [arkon/ng-sidebar#51]
Hi,

I tried to implement sidebar inside the child component (an div container) without success.
Is this possible? Do you have an example how to implement this?

Thanks

**Comment by arkon** [comment]
You should be using the provided `<ng-sidebar-container>` component. Check out the example [in the README](https://github.com/arkon/ng-sidebar#usage).

**Comment by Olmer1st** [comment]
I used this component and the sidebar still opened on all page instead small parent div.

**Comment by Olmer1st** [comment]
Like here:
http://dpiccone.github.io/ng-pageslide/examples/ (angular 1)
In a container or In a container with directions

**Comment by arkon** [comment]
Ah, I see what you mean. You might have to use CSS to do it, like so:

```css
.your_container {
  position: relative;
}

.ng-sidebar {
  position: absolute;
}
```

**Comment by Olmer1st** [comment]
Thanks. I'll check and inform you if it work shortly.

**Comment by vinsticSmith** [comment]
I tried and failed too. It's very useful to have such feature. Anyway, thanks a lot.

**Comment by Olmer1st** [comment]
I have checked: 
it work with  left and right.
top and bottom - not work as required.

**Comment by janigro** [comment]
I have found that **ng-sidebar** has many styles that you cannot simply overwrite (such as background-color). So far, I only found it possible to do it by adding '!Important' to it. Maybe I am missing something very obvious (there's a lot I don't know/understand about css).

You could try:

`
.ng-sidebar {
  position: absolute !Important;
}`

Apparently, these hard to set properties are being inserted directly into the body, taking more priority than your styles.css file. Maybe I'm saying no sense, as I am new to css. However, the empirical fact remains: adding !Important, worked for me (in setting the background color of the sidebar).

**Comment by Olmer1st** [comment]
@janigro i'm using sidebar in my own component. 
position left and right work adding `:host /deep/ .ng-sidebar { position: absolute; }`

**Comment by janigro** [comment]
Did not know the :host /deep/ trick! Thanks, it also worked for changing the sidebar background-color!

**Comment by arkon** [comment]
Yes, sorry. That seems to be a common source of confusion due to how Angular does things. I've added a note in the README to explain it: https://github.com/arkon/ng-sidebar#styling

**Comment by tuurbo** [comment]
When adding```position: absolute``` to ```.ng-sidebar``` and ```position: relative``` to ```<ng-sidebar-container>``` it adds a horizontal scroll bar to the entire page. Any suggestions?

**Comment by Sifawm** [comment]
Another issue with this question is that the modes on ng-sidebar not works correctly. I have prepared a pull request to fix this.

The proposed solution is to have a registry function on the SidebarService. The SidebarComponent registers on it and SidebarContainerComponent subscribes to this elements.

**Comment by Olmer1st** [comment]
Hi,
after upgrading to 5.0.1 the possible workarounds stopped work.
Currently restored to old 4.2.3

**Comment by arkon** [comment]
@Olmer1st That's likely because v5.0.0 switched from `ViewEncapsulation.None` to the default value (`ViewEncapsulation.Emulated`), so you'd likely need to be more specific with the CSS selectors.

**Comment by Olmer1st** [comment]
@arkon And what does it mean in our case? Can you provide an example?

**Comment by arkon** [comment]
@Olmer1st What did you have before as workarounds?

**Comment by Olmer1st** [comment]
@arkon  workaround:
`:host >>> .ng-sidebar { position: absolute; }
:host >>> .ng-sidebar--opened {
    position: relative;
}
`

**Comment by arkon** [comment]
@Olmer1st This is sort of annoying, but unfortunately the solution is to just add `!important` to the rules.

**Comment by Ketec** [comment]
Seems like something major did change, completely broken aswell now. Instead of staying inside the container, it now stretches  and overflow: hidden does nothing now (i used it to show a scrollbar if the content was taller then the container it was in).

**Comment by Ketec** [comment]
I had similar issue - had to wipe node_modules - apparently there were changes that yarn couldn't properly clean/update.

**Comment by arkon** [comment]
This is addressed in v6.0.0 and _should_ work as you would expect. Let me know if it's not what you were expecting.
